src/huggingface_hub/hf_api.py

```diff
@@ -170,6 +171,57 @@ class BlobLfsInfo(TypedDict, total=False):
     sha256: str
 
 
+@dataclass
+class CommitInfo:
+    """Data structure containing information about a newly created commit.
+
+    Returned by [`create_commit`].
+
+    Args:
+        commit_url (`str`):
+            Url where to find the commit.
+
+        commit_message (`str`):
+            The summary (first line) of the commit that has been created.
+
+        commit_description (`str`):
+            Description of the commit that has been created. Can be empty.
+
+        oid (`str`):
+            Commit hash id. Example: `"91c54ad1727ee830252e457677f467be0bfd8a57"`.
+
+        pr_url (`str`, *optional*):
+            Url to the PR that has been created, if any. Populated when `create_pr=True`
+            is passed.
+
+        pr_revision (`str`, *optional*):
+            Revision of the PR that has been created, if any. Populated when
+            `create_pr=True` is passed. Example: `"refs/pr/1"`.
+
+        pr_num (`int`, *optional*):
+            Number of the PR discussion that has been created, if any. Populated when
+            `create_pr=True` is passed. Can be passed as `discussion_num` in
+            [`get_discussion_details`]. Example: `1`.
+    """
+
+    commit_url: str
+    commit_message: str
+    commit_description: str
+    oid: str
+    pr_url: Optional[str] = None
+
+    @property
+    def pr_revision(self) -> Optional[str]:
+        if self.pr_url is not None:
+            return _parse_revision_from_pr_url(self.pr_url)
+
+    @property
+    def pr_num(self) -> Optional[int]:
+        revision = self.pr_revision
+        if revision is not None:
+            return int(revision.split("/")[-1])
+
```

Comment:
Nit: Maybe consider adding a `__str__` method for a nice string representation

Reply:
@osanseviero  Thanks for the comment , I discovered that a good way to handle properties in dataclasses is to use the `__post_init__` method and `field(init=False)` in attribute definition. This way we don't use `@property` and the dataclass is fully aware of the computed attributes as if they were normal attributes.

See https://docs.python.org/3.10/library/dataclasses.html#post-init-processing and https://github.com/huggingface/huggingface_hub/pull/1066/commits/475336f4cca15c3b2096d5be0307c6cfade1eb33.